services/management/backup/restore_history_service.go

```diff
@@ -0,0 +1,193 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package backup provides backup functionality.
+package backup
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes"
+	backupv1beta1 "github.com/percona/pmm/api/managementpb/backup"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+// RestoreHistoryService represents restore history API.
+type RestoreHistoryService struct {
+	l  *logrus.Entry
+	db *reform.DB
+}
+
+// NewRestoreHistoryService creates new restore history API service.
+func NewRestoreHistoryService(db *reform.DB) *RestoreHistoryService {
+	return &RestoreHistoryService{
+		l:  logrus.WithField("component", "management/backup/restore_history"),
+		db: db,
+	}
+}
+
+// Enabled returns if service is enabled and can be used.
+func (s *RestoreHistoryService) Enabled() bool {
+	settings, err := models.GetSettings(s.db)
+	if err != nil {
+		s.l.WithError(err).Error("can't get settings")
+		return false
+	}
+	return settings.BackupManagement.Enabled
+}
+
+// ListRestoreHistory returns a list of restore history.
+func (s *RestoreHistoryService) ListRestoreHistory(
+	context.Context,
+	*backupv1beta1.ListRestoreHistoryRequest,
+) (*backupv1beta1.ListRestoreHistoryResponse, error) {
+	q := s.db.Querier
+
+	items, err := models.FindRestoreHistoryItems(q)
+	if err != nil {
+		return nil, err
+	}
+
+	artifactIDs := make([]string, 0, len(items))
+	for _, i := range items {
+		artifactIDs = append(artifactIDs, i.ArtifactID)
+	}
+	artifacts, err := models.FindArtifactsByIDs(q, artifactIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	locationIDs := make([]string, 0, len(items))
+	for _, a := range artifacts {
+		locationIDs = append(locationIDs, a.LocationID)
+	}
+	locations, err := models.FindBackupLocationsByIDs(q, locationIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	serviceIDs := make([]string, 0, len(items))
+	for _, b := range items {
+		serviceIDs = append(serviceIDs, b.ServiceID)
+	}
+
+	services, err := models.FindServicesByIDs(q, serviceIDs)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Let's wrap all this stuff in transaction

Reply:
done